docs/plugins/reindex.asciidoc

```diff
@@ -0,0 +1,596 @@
+its worth it. If possible, prefer a more selective query to `size` and `sort`.
+This will copy 10000 documents from `twitter` into `new_twitter`:
+
+[source,js]
+--------------------------------------------------
+POST /_index_by_search
+{
+  "size": 10000,
+  "src": {
+    "index": "twitter",
+    "sort": { "date": "desc" }
+  },
+  "dest": {
+    "index": "new_twitter"
+  }
+}
+--------------------------------------------------
+// AUTOSENSE
+
+Like `_update_by_query`, `_reindex` supports a script that modifies the
+document. Unlike `_update_by_query`, the script is allowed to modify the
+document's metadata. This example bumps the version of the source document:
+
+[source,js]
+--------------------------------------------------
+POST /_index_by_search
+{
+  "src": {
+    "index": "twitter",
+  },
+  "dest": {
+    "index": "new_twitter",
+    "version_type": "external"
+  }
+  "script": {
+    "internal": "if (ctx._source.foo == 'bar') {ctx._version++; ctx._source.remove('foo')}"
+  }
+}
+--------------------------------------------------
+// AUTOSENSE
+
+Think of the posibilities! Just be careful! With great power.... Anyway! You
+can also change the "_type" and "_id" and even "_index" to change the
+destination as much as you need.
+
+Another thing about `version`, setting it to `null` or clearing it from the
+`ctx` map is just like not sending the version in an indexing request: it will
+cause that document to be overwritten in the target index regardless of the
+versioning there.
+
+By default if `_reindex` sees a document with routing then the routing is
+preserved unless its changed by scripting. You can set `routing` on the `dest`
+request to changes this:
+
+`keep`::
+
+Sets the routing on the bulk request sent for each match to the routing on
+the match. The default.
+
+`discard`::
+
+Sets the routing on the bulk request sent for each match to null.
+
+`=<some text>`::
+
+Sets the routing on the bulk request sent for each match to all text after
+the `=`.
+
+For example, you can use the following query to copy all documents from
+the `src` index with the company name `cat` into the `dest` index with
+routing set to `cat`.
+[source,js]
+--------------------------------------------------
+POST /_reindex
+{
+  "src": {
+    "index": "src"
+    "query": {
+      "match": {
+        "company": "cat"
+      }
+    }
+  }
+  "index": {
+    "index": "dest",
+    "routing": "=cat"
+  }
+}
+--------------------------------------------------
+// AUTOSENSE
+
+
+[float]
+=== URL Parameters
+
+In addition to the standard parameters like `pretty`, all APIs in this plugin
+support `refresh`, `consistency`, and `timeout`.
+
+Sending the `refresh` url parameter will cause all indexes to which the request
+wrote to be refreshed. This is different than the Index API's `refresh`
+parameter which causes just the shard that received the new data to be indexed.
+
+`consistency` controls how many copies of a shard must response to the write
```

Comment:
Probably also link to the [reference docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#index-consistency) so the user knows which values are possible (i.e. one, all, quorum)?